src/main/java/org/openstreetmap/atlas/checks/utility/OpenStreetMapCheckFlagConverter.java

```diff
@@ -0,0 +1,209 @@
+        {
+            final String osmIdentifier = object.getUniqueIdentifier().replaceFirst("\\d{6}$",
+                    CommonConstants.EMPTY_STRING);
+            objectsMap.putIfAbsent(osmIdentifier, new HashSet<>());
+            objectsMap.get(osmIdentifier).add(object);
+        });
+
+        // Map feature changes by their unique osm id
+        final Map<String, Set<FeatureChange>> suggestionMap = new HashMap<>();
+        flag.getFixSuggestions().forEach(suggestion ->
+        {
+            final String osmIdentifier = String.format("%s%s",
+                    StringUtils.capitalize(suggestion.getBeforeView().getType().toString()),
+                    suggestion.getBeforeView().getOsmIdentifier());
+            suggestionMap.putIfAbsent(osmIdentifier, new HashSet<>());
+            suggestionMap.get(osmIdentifier).add(suggestion);
+        });
+
+        try
+        {
+            // Re-add points
+            objectsMap.entrySet().stream()
+                    .filter(entry -> entry.getKey().equalsIgnoreCase(NULL_IDENTIFIERS))
+                    .flatMap(entry -> entry.getValue().stream()).forEach(newFlag::addObject);
+            // Convert and add atlas based features
+            convertFlaggedObjects(objectsMap, newFlag);
+            // Convert and add suggestions
+            convertFixSuggestions(suggestionMap, newFlag);
+        }
+        catch (final CoreException exception)
+        {
+            return Optional.empty();
+        }
+
+        return Optional.of(newFlag);
+    }
+
+    /**
+     * Merge {@link FeatureChange} fix suggestions to make them OSMified.
+     *
+     * @param suggestionMap
+     *            {@link Map} of {@link FeatureChange}s by unique OSM ID
+     * @param newFlag
+     *            {@link CheckFlag} to add the converted suggestions to
+     */
+    private static void convertFixSuggestions(final Map<String, Set<FeatureChange>> suggestionMap,
+            final CheckFlag newFlag)
+    {
+        // Go through each entry
+        suggestionMap.entrySet().stream().flatMap(entry ->
+        {
+            // Use the first feature change as a base for the merged one
+            final FeatureChange firstFeatureChange = entry.getValue().iterator().next();
+            switch (entry.getValue().iterator().next().getBeforeView().getType())
+            {
+                // Merge edge FeatureChanges by concatenating the geometries of the before and after
+                // views and applying them to a copy of the first feature change
+                case EDGE:
+                    // Sort the before views and concatenate their geometries
+                    final Optional<PolyLine> concatenatedBeforePolyline = entry.getValue().stream()
+                            .map(FeatureChange::getBeforeView)
+                            .sorted(Comparator.comparing(AtlasEntity::getIdentifier))
+                            .map(entity -> ((Edge) entity).asPolyLine()).reduce(PolyLine::append);
```

Comment:
"concatenate" when it comes to Polylines really means merging them together, correct?

Reply:
yes